target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-xx8300.dtsi

```diff
@@ -0,0 +1,319 @@
+
+};
+
+&cryptobam {
+	status = "okay";
+};
+
+&nand {
+	status = "okay";
+
+	pinctrl-0 = <&nand_pins>;
+	pinctrl-names = "default";
+
+	nand@0 {
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "sbl1";
+				reg = <0x0 0x100000>;
+				read-only;
+			};
+
+			partition@100000 {
+				label = "mibib";
+				reg = <0x100000 0x100000>;
+				read-only;
+			};
+
+			partition@200000 {
+				label = "qsee";
+				reg = <0x200000 0x100000>;
+				read-only;
+			};
+
+			partition@300000 {
+				label = "cdt";
+				reg = <0x300000 0x80000>;
+				read-only;
+			};
+
+			partition@380000 {
+				label = "appsblenv";
+				reg = <0x380000 0x80000>;
+				read-only;
+			};
+
+			partition@400000 {
+				label = "ART";
+				reg = <0x400000 0x80000>;
+				read-only;
+			};
+
+			partition@480000 {
+				label = "appsbl";
+				reg = <0x480000 0x200000>;
+				read-only;
+			};
+
+			partition@680000 {
+				label = "u_env";
+				reg = <0x680000 0x80000>;
+				// writable -- U-Boot environment
+			};
+
+			partition@700000 {
+				label = "s_env";
+				reg = <0x700000 0x40000>;
+				// writable -- Boot counter records
+			};
+
+			partition@740000 {
+				label = "devinfo";
+				reg = <0x740000 0x40000>;
+				read-only;
+			};
+
+			partition@780000 {
+				label = "kernel";
+				reg = <0x780000 0x5800000>;
+			};
+
+			partition@a80000 {
+				label = "rootfs";
+				reg = <0xa80000 0x5500000>;
+			};
+
+			partition@5f80000 {
+				label = "alt_kernel";
+				reg = <0x5f80000 0x5800000>;
+			};
+
+			partition@6280000 {
+				label = "alt_rootfs";
+				reg = <0x6280000 0x5500000>;
+			};
+
+			partition@b780000 {
+				label = "sysdiag";
+				reg = <0xb780000 0x100000>;
+				read-only;
+			};
+
+			partition@b880000 {
+				label = "syscfg";
+				reg = <0xb880000 0x4680000>;
+				read-only;
+			};
+		};
+	};
+};
+
+&pcie0 {
+	status = "okay";
+
+	perst-gpio = <&tlmm 38 GPIO_ACTIVE_LOW>;
+	wake-gpio = <&tlmm 50 GPIO_ACTIVE_LOW>;
+
+	bridge@0,0 {
+		reg = <0x00000000 0 0 0 0>;
+		#address-cells = <3>;
+		#size-cells = <2>;
+		ranges;
+
+		wifi2: wifi@1,0 {
+			compatible = "qcom,ath10k";
+			reg = <0x00010000 0 0 0 0>;
+		};
+	};
+};
+
+&qpic_bam {
+	status = "okay";
+};
+
+&tlmm {
+	serial_0_pins: serial0-pinmux {
+		pins = "gpio16", "gpio17";
+		function = "blsp_uart0";
+		bias-disable;
+	};
+
+	nand_pins: nand_pins {
+		pullups {
+			pins = "gpio53", "gpio58", "gpio59";
+			function = "qpic";
+			bias-pull-up;
+		};
+
+		// gpio61 controls led_usb
+
+		pulldowns {
+			pins =  "gpio55", "gpio56", "gpio57",
+				"gpio60", "gpio62", "gpio63",
+				"gpio64", "gpio65", "gpio66",
+				"gpio67", "gpio68", "gpio69";
+			function = "qpic";
+			bias-pull-down;
+		};
+	};
+};
+
+&usb2_hs_phy {
+	status = "okay";
+};
+
+&usb3_hs_phy {
+	status = "okay";
+};
+
+&usb3_ss_phy {
+	status = "okay";
+};
+
+&wifi0 {
+	status = "okay";
+	qcom,ath10k-calibration-variant = "linksys-ea8300-fcc";
```

Comment:
Hmpf. 

I downloaded the MR8300 Ver. 1.1.7.201281 and EA8300 Ver. 1.1.5.201210 firmware files. I extracted them with binwalk and looked if the board files between the MR8300 and EA8300 under ubifs/lib/firmware/IPQ4019/FCC and FCC1.1 matched... They did not. What's more the boardfiles of the V1.0 are different from V1.1 (Linksys included the V1.0 boardfiles in the V1.1 firmwares).

EA8300
057ed5528e360197c2239796710cb61c  ubifs/lib/firmware/IPQ4019/FCC/boarddata_0.bin
0e820f5946b5773ad9f4a74809c05b79  ubifs/lib/firmware/IPQ4019/FCC/boarddata_1.bin

f6a51bf88a198f38f685c2784d4349c1  ubifs/lib/firmware/IPQ4019/FCC1.1/boarddata_0.bin
50ccc062211a36131b72fedf89d14689  ubifs/lib/firmware/IPQ4019/FCC1.1/boarddata_1.bin

bce8860a0812900da671b0bf67ff4630  ubifs/lib/firmware/QCA9888/FCC1.1/boarddata_0.bin
62049f88a498e51870973143e376e7af  ubifs/lib/firmware/QCA9888/FCC/boarddata_0.bin

vs

MR8300
1e0be7672bd83ccad24cf8e998b8d1ef  ubifs/lib/firmware/IPQ4019/FCC/boarddata_0.bin
5a65c2d79f838a6e0d9d05b776491ac8  ubifs/lib/firmware/IPQ4019/FCC/boarddata_1.bin

ec1599628dfd887c8a3960cce3cf1e90  ubifs/lib/firmware/IPQ4019/FCC1.1/boarddata_0.bin
90bf41592d46714de96465f8d100d32c  ubifs/lib/firmware/IPQ4019/FCC1.1/boarddata_1.bin

1f263c52e7862bccab6fd6551092077c  ubifs/lib/firmware/QCA9888/FCC/boarddata_0.bin
b59312f762d5d16f1f7231c36d2706e2  ubifs/lib/firmware/QCA9888/FCC1.1/boarddata_0.bin

Can you please use a separate variant identifer for the MR8300 and prepare the ipq-wifi packages and post the boardfiles to the ath10k-devel mailing list (as per: https://wireless.wiki.kernel.org/en/users/drivers/ath10k/boardfiles).
I don't want to repeat 81adb132daa529167cc4df9849882ca144e7cba7 .

Reply:
This is going to be a bit of homework for me, but I am up to the challenge if you have the patience to work with me. (I have never done this before :-) )

So, I think for a variant identifier, we could properly follow what was done for the ea8300 and make this one the "linksys-mr8300-fcc" I would think.

I have never prepared a ipq-wifi package nor submitted board files up the dev mailing list, though I think we can do that :-) Would you be kind enough to outline in a bit more detail what needs to be done and I will do my best to make sure it happens.

Reply:
What revision of the MR8300 did you get? Is it the V1.0 or V1.1? (I can skip the other revision)

The individual boardfiles (raw board.bin - only contains calibration for a particular wifi) are located
in the vendor firmware. binwalk can extract the FW_MR8300_1.1.7.201281_prod.img and you'll find
the files in ubifs-root-directoy /lib/firmware/IPQ4019 and QCA9888. 

Once you have these, you can package them to a board-2.bin (this is sort of a library of multiple board.bin with IDs)
and run tests which works best.
You can follow fa03d441e96eeec0781ccbe757a3641c9dcec785 suggestion:

    To create the individual board-2.bin: Use the ath10k-bdencoder
    utility from the qca-swiss-army-knife repository:
    <https://github.com/qca/qca-swiss-army-knife>
    The raw board.bin files have to be extracted from the
    vendor's source GPL.tar archieves.

ath10k-bdecoder can do both: create or extract board-2.bin. I think you could just grab the
EA8300's board-linksys_ea8300.[qca4019,qca9888] files from openwrt's package/firmware/ipq-wifi.
Extract them with ath10k-bdecoder (each in their own folder).
Then edit the generated .json file (swap EA8300 with MR8300) and rename the MR8300 boardfiles
you got from the FW_MR8300...img image to match the new "MR8300" file names in the .json.
Then rebuild the board_2.bin and rename it board-linksys_mr8300-v1.[0 or 1 - depending on what you have].[qca4019,qca9888] and wire it up into the ipq-wifi makefile. edit the wifi nodes in your MR8300 .dts and
add the ipq-wifi packages to your image.

For the submission to the ath10k mailing-list. You start by copying the provided template on the wiki.
Run md5sum on the individual .bin files and attach the renamed individual .bin files
[not the complete board-2.bin].

Reply:
Perfect, this gives me enough to get started.

Reply:
> What revision of the MR8300 did you get? Is it the V1.0 or V1.1? (I can skip the other revision)

I have 3 x units, and they are all revision 1.0. There are no distinct markings on the packaging or label to state v1.1, which Linksys normally does.

Reply:
@chunkeey  with your excellent advice above, I was able to complete what you asked. I pulled down the existing ea8300 qca4019 and qca9888 bin files, I extracted them, made the modifications to the board-2.json, brought in the corresponding boarddata_0 and 1 .bin files from FCC dir's from Linksys firmware, renamed them appropriately and recreated the new board-2.bin file.

Here is an outline of the files I have now changed and pushed to this PR:
![image](https://user-images.githubusercontent.com/22136734/93029073-8a4ff300-f5e6-11ea-9746-e37fd7ef2ad4.png)

I have compiled a fresh image generate from make menuconfig and selecting MR8300 as target, and the compilation was a success.

Clean Kernel Boot log:
```
[    0.000000] Booting Linux on physical CPU 0x0
[    0.000000] Linux version 5.4.63 (openwrt@debian-openwrt) (gcc version 8.4.0 (OpenWrt GCC 8.4.0 r14443+2-b691e0b255)) #0 SMP Fri Sep 11 23:56:46 2020
[    0.000000] CPU: ARMv7 Processor [410fc075] revision 5 (ARMv7), cr=10c5387d
[    0.000000] CPU: div instructions available: patching division code
[    0.000000] CPU: PIPT / VIPT nonaliasing data cache, VIPT aliasing instruction cache
[    0.000000] OF: fdt: Machine model: Linksys MR8300 (Dallas)
[    0.000000] Memory policy: Data cache writealloc
[    0.000000] On node 0 totalpages: 130560
[    0.000000]   Normal zone: 1152 pages used for memmap
[    0.000000]   Normal zone: 0 pages reserved
[    0.000000]   Normal zone: 130560 pages, LIFO batch:31
[    0.000000] percpu: Embedded 15 pages/cpu s30092 r8192 d23156 u61440
[    0.000000] pcpu-alloc: s30092 r8192 d23156 u61440 alloc=15*4096
[    0.000000] pcpu-alloc: [0] 0 [0] 1 [0] 2 [0] 3 
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 129408
[    0.000000] Kernel command line: init=/sbin/init rootfstype=ubifs ubi.mtd=13,2048 root=ubi0:ubifs rootwait rw root=/dev/ubiblock0_0 rootfstype=squashfs ro
[    0.000000] Dentry cache hash table entries: 65536 (order: 6, 262144 bytes, linear)
[    0.000000] Inode-cache hash table entries: 32768 (order: 5, 131072 bytes, linear)
[    0.000000] mem auto-init: stack:off, heap alloc:off, heap free:off
[    0.000000] Memory: 508280K/522240K available (5441K kernel code, 183K rwdata, 1440K rodata, 1024K init, 231K bss, 13960K reserved, 0K cma-reserved, 0K highmem)
[    0.000000] SLUB: HWalign=64, Order=0-3, MinObjects=0, CPUs=4, Nodes=1
[    0.000000] rcu: Hierarchical RCU implementation.
[    0.000000] rcu: RCU calculated value of scheduler-enlistment delay is 10 jiffies.
[    0.000000] NR_IRQS: 16, nr_irqs: 16, preallocated irqs: 16
[    0.000000] random: get_random_bytes called from start_kernel+0x2f0/0x498 with crng_init=0
[    0.000000] arch_timer: cp15 timer(s) running at 48.00MHz (virt).
[    0.000000] clocksource: arch_sys_counter: mask: 0xffffffffffffff max_cycles: 0xb11fd3bfb, max_idle_ns: 440795203732 ns
[    0.000007] sched_clock: 56 bits at 48MHz, resolution 20ns, wraps every 4398046511096ns
[    0.000020] Switching to timer-based delay loop, resolution 20ns
[    0.000234] Calibrating delay loop (skipped), value calculated using timer frequency.. 96.00 BogoMIPS (lpj=480000)
[    0.000250] pid_max: default: 32768 minimum: 301
[    0.000452] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.000471] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.001455] CPU: Testing write buffer coherency: ok
[    0.002486] Setting up static identity map for 0x80300000 - 0x80300060
[    0.002623] rcu: Hierarchical SRCU implementation.
[    0.003067] smp: Bringing up secondary CPUs ...
[    0.006149] smp: Brought up 1 node, 4 CPUs
[    0.006168] SMP: Total of 4 processors activated (384.00 BogoMIPS).
[    0.006176] CPU: All CPU(s) started in SVC mode.
[    0.010894] VFP support v0.3: implementor 41 architecture 2 part 30 variant 7 rev 5
[    0.011058] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.011084] futex hash table entries: 1024 (order: 4, 65536 bytes, linear)
[    0.011343] pinctrl core: initialized pinctrl subsystem
[    0.012409] NET: Registered protocol family 16
[    0.012781] DMA: preallocated 256 KiB pool for atomic coherent allocations
[    0.014010] cpuidle: using governor ladder
[    0.014059] cpuidle: using governor menu
[    0.034900] cryptd: max_cpu_qlen set to 1000
[    0.038287] usbcore: registered new interface driver usbfs
[    0.038355] usbcore: registered new interface driver hub
[    0.038443] usbcore: registered new device driver usb
[    0.038503] pps_core: LinuxPPS API ver. 1 registered
[    0.038513] pps_core: Software ver. 5.3.6 - Copyright 2005-2007 Rodolfo Giometti <giometti@linux.it>
[    0.038538] PTP clock support registered
[    0.039237] workqueue: max_active 576 requested for napi_workq is out of range, clamping between 1 and 512
[    0.040379] clocksource: Switched to clocksource arch_sys_counter
[    0.041216] thermal_sys: Registered thermal governor 'step_wise'
[    0.041507] NET: Registered protocol family 2
[    0.042100] tcp_listen_portaddr_hash hash table entries: 512 (order: 0, 6144 bytes, linear)
[    0.042145] TCP established hash table entries: 4096 (order: 2, 16384 bytes, linear)
[    0.042205] TCP bind hash table entries: 4096 (order: 3, 32768 bytes, linear)
[    0.042292] TCP: Hash tables configured (established 4096 bind 4096)
[    0.042406] UDP hash table entries: 256 (order: 1, 8192 bytes, linear)
[    0.042444] UDP-Lite hash table entries: 256 (order: 1, 8192 bytes, linear)
[    0.042674] NET: Registered protocol family 1
[    0.042713] PCI: CLS 0 bytes, default 64
[    0.044700] workingset: timestamp_bits=14 max_order=17 bucket_order=3
[    0.050503] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.050518] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    0.102003] qcom-pcie 40000000.pci: host bridge /soc/pci@40000000 ranges:
[    0.102054] qcom-pcie 40000000.pci:    IO 0x40200000..0x402fffff -> 0x40200000
[    0.102084] qcom-pcie 40000000.pci:   MEM 0x40300000..0x40ffffff -> 0x40300000
[    0.381953] qcom-pcie 40000000.pci: Link up
[    0.382102] qcom-pcie 40000000.pci: PCI host bridge to bus 0000:00
[    0.382122] pci_bus 0000:00: root bus resource [bus 00-ff]
[    0.382140] pci_bus 0000:00: root bus resource [io  0x0000-0xfffff] (bus address [0x40200000-0x402fffff])
[    0.382154] pci_bus 0000:00: root bus resource [mem 0x40300000-0x40ffffff]
[    0.382195] pci 0000:00:00.0: [17cb:1001] type 01 class 0x060400
[    0.382234] pci 0000:00:00.0: reg 0x10: [mem 0x00000000-0x00000fff]
[    0.382308] pci 0000:00:00.0: PME# supported from D0 D3hot
[    0.383840] PCI: bus0: Fast back to back transfers disabled
[    0.384069] pci 0000:01:00.0: [168c:0056] type 00 class 0x028000
[    0.384283] pci 0000:01:00.0: reg 0x10: [mem 0x00000000-0x001fffff 64bit]
[    0.385005] pci 0000:01:00.0: PME# supported from D0 D3hot
[    0.386688] PCI: bus1: Fast back to back transfers disabled
[    0.386739] pci 0000:00:00.0: BAR 8: assigned [mem 0x40400000-0x405fffff]
[    0.386758] pci 0000:00:00.0: BAR 0: assigned [mem 0x40300000-0x40300fff]
[    0.386780] pci 0000:01:00.0: BAR 0: assigned [mem 0x40400000-0x405fffff 64bit]
[    0.386857] pci 0000:00:00.0: PCI bridge to [bus 01-ff]
[    0.386874] pci 0000:00:00.0:   bridge window [mem 0x40400000-0x405fffff]
[    0.389659] pcieport 0000:00:00.0: AER: enabled with IRQ 94
[    0.391495] bam-dma-engine 8e04000.dma: num-channels unspecified in dt
[    0.391517] bam-dma-engine 8e04000.dma: num-ees unspecified in dt
[    0.392839] tcsr 1949000.tcsr: setting wifi_glb_cfg = 41000000
[    0.392933] tcsr 194b000.tcsr: setting usb hs phy mode select = e700e7
[    0.393020] tcsr 1953000.ess_tcsr: setting ess interface select = 0
[    0.393104] tcsr 1957000.tcsr: setting wifi_noc_memtype_m0_m2 = 2222222
[    0.393430] Serial: 8250/16550 driver, 2 ports, IRQ sharing disabled
[    0.393986] msm_serial 78af000.serial: msm_serial: detected port #0
[    0.394028] msm_serial 78af000.serial: uartclk = 1843200
[    0.394081] 78af000.serial: ttyMSM0 at MMIO 0x78af000 (irq = 24, base_baud = 115200) is a MSM
[    0.394107] msm_serial: console setup on port #0
[    1.022514] printk: console [ttyMSM0] enabled
[    1.027432] msm_serial: driver initialized
[    1.036449] loop: module loaded
[    1.038059] nand: device found, Manufacturer ID: 0xef, Chip ID: 0xda
[    1.038458] nand: Winbond W29N02GV
[    1.045106] nand: 256 MiB, SLC, erase size: 128 KiB, page size: 2048, OOB size: 64
[    1.048367] 16 fixed-partitions partitions found on MTD device qcom_nand.0
[    1.055852] Creating 16 MTD partitions on "qcom_nand.0":
[    1.062710] 0x000000000000-0x000000100000 : "sbl1"
[    1.070063] 0x000000100000-0x000000200000 : "mibib"
[    1.074611] 0x000000200000-0x000000300000 : "qsee"
[    1.079385] 0x000000300000-0x000000380000 : "cdt"
[    1.083784] 0x000000380000-0x000000400000 : "appsblenv"
[    1.088520] 0x000000400000-0x000000480000 : "ART"
[    1.093589] 0x000000480000-0x000000680000 : "appsbl"
[    1.099876] 0x000000680000-0x000000700000 : "u_env"
[    1.103501] 0x000000700000-0x000000740000 : "s_env"
[    1.107821] 0x000000740000-0x000000780000 : "devinfo"
[    1.112791] 0x000000780000-0x000005f80000 : "kernel"
[    1.125470] random: fast init done
[    1.202727] 0x000000a80000-0x000005f80000 : "rootfs"
[    1.285720] mtd: device 11 (rootfs) set to be root filesystem
[    1.286065] mtdsplit: no squashfs found in "rootfs"
[    1.290511] 0x000005f80000-0x00000b780000 : "alt_kernel"
[    1.381614] 0x000006280000-0x00000b780000 : "alt_rootfs"
[    1.465129] 0x00000b780000-0x00000b880000 : "sysdiag"
[    1.467028] 0x00000b880000-0x00000ff00000 : "syscfg"
[    1.539779] libphy: ipq40xx_mdio: probed
[    1.571238] ESS reset ok!
[    1.604205] ESS reset ok!
[    2.022302] libphy: Fixed MDIO Bus: probed
[    2.024069] EDMA using MAC@ - using
[    2.024080] 32:93:07:9a:33:e5
[    2.031398] i2c /dev entries driver
[    2.033767] sdhci: Secure Digital Host Controller Interface driver
[    2.035097] sdhci: Copyright(c) Pierre Ossman
[    2.041380] sdhci-pltfm: SDHCI platform and OF driver helper
[    2.049162] NET: Registered protocol family 10
[    2.052872] Segment Routing with IPv6
[    2.055837] NET: Registered protocol family 17
[    2.059531] 8021q: 802.1Q VLAN Support v1.8
[    2.063997] Registering SWP/SWPB emulation handler
[    2.077404] ubi0: attaching mtd13
[    2.102741] UBI: EOF marker found, PEBs from 35 will be erased
[    2.202732] ubi0: scanning is finished
[    2.228040] ubi0: volume 1 ("rootfs_data") re-sized from 9 to 603 LEBs
[    2.228846] ubi0: attached mtd13 (name "alt_rootfs", size 85 MiB)
[    2.233551] ubi0: PEB size: 131072 bytes (128 KiB), LEB size: 126976 bytes
[    2.239635] ubi0: min./max. I/O unit sizes: 2048/2048, sub-page size 2048
[    2.246437] ubi0: VID header offset: 2048 (aligned 2048), data offset: 4096
[    2.253288] ubi0: good PEBs: 680, bad PEBs: 0, corrupted PEBs: 0
[    2.260029] ubi0: user volume: 2, internal volumes: 1, max. volumes count: 128
[    2.266312] ubi0: max/mean erase counter: 1/0, WL threshold: 4096, image sequence number: 1599868606
[    2.273346] ubi0: available PEBs: 0, total reserved PEBs: 680, PEBs reserved for bad PEB handling: 40
[    2.282660] ubi0: background thread "ubi_bgt0d" started, PID 103
[    2.284539] block ubiblock0_0: created from ubi0:0(rootfs)
[    2.297826] hctosys: unable to open rtc device (rtc0)
[    2.313904] VFS: Mounted root (squashfs filesystem) readonly on device 254:0.
[    2.315642] Freeing unused kernel memory: 1024K
[    2.340806] Run /sbin/init as init process
[    2.827349] init: Console is alive
[    2.827654] init: - watchdog -
[    2.961126] random: crng init done
[    3.587534] kmodloader: loading kernel modules from /etc/modules-boot.d/*
[    3.685030] dwc3-qcom 8af8800.usb3: IRQ hs_phy_irq not found
[    3.685067] dwc3-qcom 8af8800.usb3: IRQ dp_hs_phy_irq not found
[    3.689761] dwc3-qcom 8af8800.usb3: IRQ dm_hs_phy_irq not found
[    3.695463] dwc3-qcom 8af8800.usb3: IRQ ss_phy_irq not found
[    3.703086] dwc3-qcom 60f8800.usb2: IRQ hs_phy_irq not found
[    3.707211] dwc3-qcom 60f8800.usb2: IRQ dp_hs_phy_irq not found
[    3.712893] dwc3-qcom 60f8800.usb2: IRQ dm_hs_phy_irq not found
[    3.718495] dwc3-qcom 60f8800.usb2: IRQ ss_phy_irq not found
[    3.731346] dwc3 8a00000.dwc3: Failed to get clk 'ref': -2
[    3.895777] dwc3 6000000.dwc3: Failed to get clk 'ref': -2
[    4.033789] xhci-hcd xhci-hcd.0.auto: xHCI Host Controller
[    4.033868] xhci-hcd xhci-hcd.0.auto: new USB bus registered, assigned bus number 1
[    4.038513] xhci-hcd xhci-hcd.0.auto: hcc params 0x0228f665 hci version 0x100 quirks 0x0000000002010010
[    4.045952] xhci-hcd xhci-hcd.0.auto: irq 97, io mem 0x08a00000
[    4.056234] hub 1-0:1.0: USB hub found
[    4.065841] hub 1-0:1.0: 1 port detected
[    4.066229] xhci-hcd xhci-hcd.0.auto: xHCI Host Controller
[    4.068923] xhci-hcd xhci-hcd.0.auto: new USB bus registered, assigned bus number 2
[    4.074253] xhci-hcd xhci-hcd.0.auto: Host supports USB 3.0 SuperSpeed
[    4.081916] usb usb2: We don't know the algorithms for LPM for this host, disabling LPM.
[    4.088927] hub 2-0:1.0: USB hub found
[    4.096667] hub 2-0:1.0: 1 port detected
[    4.100720] xhci-hcd xhci-hcd.1.auto: xHCI Host Controller
[    4.104263] xhci-hcd xhci-hcd.1.auto: new USB bus registered, assigned bus number 3
[    4.109761] xhci-hcd xhci-hcd.1.auto: hcc params 0x0220f665 hci version 0x100 quirks 0x0000000002010010
[    4.117202] xhci-hcd xhci-hcd.1.auto: irq 98, io mem 0x06000000
[    4.132172] hub 3-0:1.0: USB hub found
[    4.132482] hub 3-0:1.0: 1 port detected
[    4.136519] xhci-hcd xhci-hcd.1.auto: xHCI Host Controller
[    4.140276] xhci-hcd xhci-hcd.1.auto: new USB bus registered, assigned bus number 4
[    4.145615] xhci-hcd xhci-hcd.1.auto: Host supports USB 3.0 SuperSpeed
[    4.153326] usb usb4: We don't know the algorithms for LPM for this host, disabling LPM.
[    4.160545] hub 4-0:1.0: USB hub found
[    4.168001] hub 4-0:1.0: config failed, hub doesn't have any ports! (err -19)
[    4.173578] kmodloader: done loading kernel modules from /etc/modules-boot.d/*
[    4.187861] init: - preinit -
[    5.206234] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
[    9.468569] UBIFS (ubi0:1): default file-system created
[    9.469184] UBIFS (ubi0:1): Mounting in unauthenticated mode
[    9.472853] UBIFS (ubi0:1): background thread "ubifs_bgt0_1" started, PID 190
[    9.546423] UBIFS (ubi0:1): UBIFS: mounted UBI device 0, volume 1, name "rootfs_data"
[    9.546476] UBIFS (ubi0:1): LEB size: 126976 bytes (124 KiB), min./max. I/O unit sizes: 2048 bytes/2048 bytes
[    9.553305] UBIFS (ubi0:1): FS size: 75296768 bytes (71 MiB, 593 LEBs), journal size 3809280 bytes (3 MiB, 30 LEBs)
[    9.563180] UBIFS (ubi0:1): reserved for root: 3556452 bytes (3473 KiB)
[    9.573435] UBIFS (ubi0:1): media format: w5/r0 (latest is w5/r0), UUID BE4F88BF-C75E-45C2-8504-2F4BD5E6C197, small LPT model
[    9.581936] mount_root: overlay filesystem has not been fully initialized yet
[    9.591894] mount_root: switching to ubifs overlay
[    9.608535] overlayfs: upper fs does not support xattr, falling back to index=off and metacopy=off.
[    9.611117] urandom-seed: Seed file not found (/etc/urandom.seed)
[    9.735189] procd: - early -
[    9.735299] procd: - watchdog -
[   10.333015] procd: - watchdog -
[   10.335916] procd: - ubus -
[   10.473357] procd: - init -
[   10.996422] kmodloader: loading kernel modules from /etc/modules.d/*
[   11.021693] urngd: v1.0.2 started.
[   11.072717] Loading modules backported from Linux version v5.8-0-gbcf876870b95
[   11.072750] Backport generated by backports.git v5.8-1-0-g79400d9e
[   11.092326] xt_time: kernel timezone is -0000
[   11.134542] PPP generic driver version 2.4.2
[   11.135542] NET: Registered protocol family 24
[   11.158593] ath10k 5.4 driver, optimized for CT firmware, probing pci device: 0x56.
[   11.158720] ath10k_mac_create, priv_size: 868  hw: e81dd2c8  hw->priv: 8c39a8f8
[   11.165670] ath10k_pci 0000:01:00.0: enabling device (0140 -> 0142)
[   11.173113] ath10k_pci 0000:01:00.0: pci irq msi oper_irq_mode 2 irq_mode 0 reset_mode 0
[   12.510305] ath10k_pci 0000:01:00.0: qca9888 hw2.0 target 0x01000000 chip_id 0x00000000 sub 0000:0000
[   12.510352] ath10k_pci 0000:01:00.0: kconfig debug 0 debugfs 1 tracing 0 dfs 1 testmode 0
[   12.522399] ath10k_pci 0000:01:00.0: firmware ver 10.4b-ct-9888-fW-13-8c5b2baa2 api 5 features mfp,peer-flow-ctrl,txstatus-noack,wmi-10.x-CT,ratemask-CT,regdump-CT,txrate-CT,flush-all-CT,pingpong-CT,ch-regs-CT,nop-CT,set-special-CT,tx-rc-CT,cust-stats-CT,txrate2-CT,beacon-cb-CT,wmi-block-ack-CT,wmi-bcn-rc-CT crc32 cbf49903
[   12.838441] ath10k_pci 0000:01:00.0: board_file api 2 bmi_id 0:23 crc32 27a6cbfc
[   14.872944] ath10k_pci 0000:01:00.0: unsupported HTC service id: 1536
[   14.873869] ath10k_pci 0000:01:00.0: 10.4 wmi init: vdevs: 16  peers: 48  tid: 96
[   14.878380] ath10k_pci 0000:01:00.0: msdu-desc: 2500  skid: 32
[   14.933878] ath10k_pci 0000:01:00.0: wmi print 'P 48/48 V 16 K 144 PH 176 T 186  msdu-desc: 2500  sw-crypt: 0 ct-sta: 0'
[   14.934776] ath10k_pci 0000:01:00.0: wmi print 'free: 114572 iram: 12804 sram: 29508'
[   15.183535] ath10k_pci 0000:01:00.0: htt-ver 2.2 wmi-op 6 htt-op 4 cal pre-cal-file max-sta 32 raw 0 hwcrypto 1
[   15.319438] ath: EEPROM regdomain: 0x834b
[   15.319448] ath: EEPROM indicates we should expect a country code
[   15.319456] ath: doing EEPROM country->regdmn map search
[   15.319463] ath: country maps to regdmn code: 0x3a
[   15.319467] ath: Country alpha2 being used: US
[   15.319471] ath: Regpair used: 0x3a
[   15.323821] ath10k_mac_create, priv_size: 924  hw: dc1694b0  hw->priv: bc285245
[   16.841855] ath10k_ahb a000000.wifi: qca4019 hw1.0 target 0x01000000 chip_id 0x003b00ff sub 0000:0000
[   16.841904] ath10k_ahb a000000.wifi: kconfig debug 0 debugfs 1 tracing 0 dfs 1 testmode 0
[   16.853424] ath10k_ahb a000000.wifi: firmware ver 10.4b-ct-4019-fW-13-8c5b2baa2 api 5 features mfp,peer-flow-ctrl,txstatus-noack,wmi-10.x-CT,ratemask-CT,regdump-CT,txrate-CT,flush-all-CT,pingpong-CT,ch-regs-CT,nop-CT,set-special-CT,tx-rc-CT,cust-stats-CT,txrate2-CT,beacon-cb-CT,wmi-block-ack-CT,wmi-bcn-rc-CT crc32 1f3ed50f
[   16.926636] ath10k_ahb a000000.wifi: board_file api 2 bmi_id 0:20 crc32 cfa24971
[   18.235019] ath10k_ahb a000000.wifi: unsupported HTC service id: 1536
[   18.236360] ath10k_ahb a000000.wifi: 10.4 wmi init: vdevs: 16  peers: 48  tid: 96
[   18.240526] ath10k_ahb a000000.wifi: msdu-desc: 2500  skid: 32
[   18.291058] ath10k_ahb a000000.wifi: wmi print 'P 48/48 V 16 K 144 PH 176 T 186  msdu-desc: 2500  sw-crypt: 0 ct-sta: 0'
[   18.292007] ath10k_ahb a000000.wifi: wmi print 'free: 53252 iram: 13576 sram: 35752'
[   18.452444] ath10k_ahb a000000.wifi: htt-ver 2.2 wmi-op 6 htt-op 4 cal pre-cal-file max-sta 32 raw 0 hwcrypto 1
[   18.609953] ath: EEPROM regdomain: 0x834b
[   18.609961] ath: EEPROM indicates we should expect a country code
[   18.609970] ath: doing EEPROM country->regdmn map search
[   18.609976] ath: country maps to regdmn code: 0x3a
[   18.609980] ath: Country alpha2 being used: US
[   18.609986] ath: Regpair used: 0x3a
[   18.613825] ath10k_mac_create, priv_size: 924  hw: f56a421f  hw->priv: eb842e42
[   19.426142] ath10k_ahb a800000.wifi: qca4019 hw1.0 target 0x01000000 chip_id 0x003b00ff sub 0000:0000
[   19.426199] ath10k_ahb a800000.wifi: kconfig debug 0 debugfs 1 tracing 0 dfs 1 testmode 0
[   19.439050] ath10k_ahb a800000.wifi: firmware ver 10.4b-ct-4019-fW-13-8c5b2baa2 api 5 features mfp,peer-flow-ctrl,txstatus-noack,wmi-10.x-CT,ratemask-CT,regdump-CT,txrate-CT,flush-all-CT,pingpong-CT,ch-regs-CT,nop-CT,set-special-CT,tx-rc-CT,cust-stats-CT,txrate2-CT,beacon-cb-CT,wmi-block-ack-CT,wmi-bcn-rc-CT crc32 1f3ed50f
[   19.490502] ath10k_ahb a800000.wifi: board_file api 2 bmi_id 0:21 crc32 cfa24971
[   20.797472] ath10k_ahb a800000.wifi: unsupported HTC service id: 1536
[   20.798422] ath10k_ahb a800000.wifi: 10.4 wmi init: vdevs: 16  peers: 48  tid: 96
[   20.802972] ath10k_ahb a800000.wifi: msdu-desc: 2500  skid: 32
[   20.853850] ath10k_ahb a800000.wifi: wmi print 'P 48/48 V 16 K 144 PH 176 T 186  msdu-desc: 2500  sw-crypt: 0 ct-sta: 0'
[   20.855399] ath10k_ahb a800000.wifi: wmi print 'free: 53252 iram: 13576 sram: 35752'
[   20.997879] ath10k_ahb a800000.wifi: htt-ver 2.2 wmi-op 6 htt-op 4 cal pre-cal-file max-sta 32 raw 0 hwcrypto 1
[   21.155226] ath: EEPROM regdomain: 0x834b
[   21.155237] ath: EEPROM indicates we should expect a country code
[   21.155245] ath: doing EEPROM country->regdmn map search
[   21.155252] ath: country maps to regdmn code: 0x3a
[   21.155256] ath: Country alpha2 being used: US
[   21.155260] ath: Regpair used: 0x3a
[   21.163403] kmodloader: done loading kernel modules from /etc/modules.d/*
[   35.551425] br-lan: port 1(eth0) entered blocking state
[   35.551478] br-lan: port 1(eth0) entered disabled state
[   35.555798] device eth0 entered promiscuous mode
[   35.573134] ess_edma c080000.edma: eth1: GMAC Link is down
[   36.550431] br-lan: port 1(eth0) entered blocking state
[   36.550466] br-lan: port 1(eth0) entered forwarding state
[   36.555577] IPv6: ADDRCONF(NETDEV_CHANGE): br-lan: link becomes ready
```

I think the final part you mentioned was I need to send an email to the mailing list. I will follow your advice and use their template. Just for confirmation, it will be these files they want and not the board-2.bin?
![image](https://user-images.githubusercontent.com/22136734/93029135-1235fd00-f5e7-11ea-8741-37937ef39c1e.png)

![image](https://user-images.githubusercontent.com/22136734/93029155-2c6fdb00-f5e7-11ea-8cdf-766162971048.png)

Reply:
I made one more push this evening, just making sure all the board files are clearly marked with -v0, as well as the reference in the make file.

Reply:
Mail has been sent to ath email group, and cc'd you @chunkeey